Frends.AzureDataLake.DeleteDirectory/Frends.AzureDataLake.DeleteDirectory.Tests/UnitTests.cs

```diff
@@ -0,0 +1,137 @@
+        "FRENDS_AZUREDATALAKE_STORAGEACCOUNT"
+    );
+    private string _containerName;
+    private string _directoryName;
+
+    [AssemblyInitialize]
+    public static void AssemblyInit(TestContext context)
+    {
+        var root = Directory.GetCurrentDirectory();
+        string projDir = Directory.GetParent(root).Parent.Parent.FullName;
+        DotEnv.Load(options: new DotEnvOptions(envFilePaths: new[] { $"{projDir}/.env.local" }));
+    }
+
+    [TestInitialize]
+    public void TestSetup()
+    {
+        // Generate unique container adn directory names to avoid conflicts when running multiple tests
+        _containerName = $"test-container{DateTime.Now.ToString("mmssffffff", CultureInfo.InvariantCulture)}";
+        _directoryName = $"test-directory{DateTime.Now.ToString("mmssffffff", CultureInfo.InvariantCulture)}";
+    }
+
+    [TestCleanup]
+    public async Task Cleanup()
+    {
+        // delete whole container after running tests
+        var client = new DataLakeServiceClient(_connectionString);
+        var container = client.GetFileSystemClient(_containerName);
+        await container.DeleteIfExistsAsync();
+    }
+
+    [TestMethod]
+    public async Task TestDeleteDirectory()
+    {
+        input = new Input { ConnectionString = _connectionString, ContainerName = _containerName, DirectoryName = _directoryName };
+        DataLakeFileSystemClient container = new DataLakeServiceClient(input.ConnectionString).GetFileSystemClient(input.ContainerName);
+        await container.CreateIfNotExistsAsync(null, new CancellationToken());
+
+        var directoryClient = AzureDataLake.GetDataLakeDirectory(input);
+        await directoryClient.CreateIfNotExistsAsync(null, new CancellationToken());
+
+        var deleted = await AzureDataLake.DeleteDirectory(input, new Options { ThrowErrorIfDirectoryDoesNotExists = true }, new CancellationToken());
+        Assert.IsTrue(deleted.DirectoryWasDeleted);
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(Exception))]
+    public async Task TestDeleteDirectory_throws_ContainerNotFound()
+    {
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = _connectionString, ContainerName = _containerName, DirectoryName = _directoryName }, new Options { ThrowErrorIfDirectoryDoesNotExists = true }, new CancellationToken());
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentNullException))]
+    public async Task TestDeleteDirectory_throws_ParameterEmpty()
+    {
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = _connectionString, ContainerName = null, DirectoryName = _directoryName }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
+        await AzureDataLake.DeleteDirectory(new Input { ConnectionString = null, ContainerName = _containerName, DirectoryName = _directoryName }, new Options { ThrowErrorIfDirectoryDoesNotExists = false }, new CancellationToken());
```

Comment:
Why the additional lines in this method - shouldn't the first line of code be already sufficient?